src/control/jobs/control_jobs.c

```diff
@@ -218,12 +258,128 @@ static float envelope(const float xx)
   }
 }
 
+static int dt_control_merge_hdr_process(dt_imageio_module_data_t *datai, const char *filename,
+                                        const void *const ivoid, void *exif, int exif_len, int imgid, int num,
+                                        int total)
+{
+  dt_control_merge_hdr_format_t *data = (dt_control_merge_hdr_format_t *)datai;
+  dt_control_merge_hdr_t *d = data->d;
+
+  dt_pthread_mutex_lock(&d->lock);
```

Comment:
I don't see why this lock is needed at all. It is never initialized, can deadlock in the error case and only guards sequential code.